src/devices/efergy_optical.c

```diff
@@ -0,0 +1,139 @@
+ * Byte 5-7: all zeros 
+ * Byte 8: Pulse Count
+ * Byte 9: sample frequency (15 seconds)
+ * Byte 10: seconds
+ * Byte 11: crc16 xmodem
+ * Byte 12: crc16 xmodem
+ * if pulse count <3 then energy =(( pulsecount/impulse-perkwh) * (3600/seconds))
+ * else energy = ((pulsecount/n_imp) * (3600/seconds))
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#include "rtl_433.h"
+#include "util.h"
+#include "data.h"
+
+static int efergy_optical_callback(bitbuffer_t *bitbuffer) {
+	unsigned num_bits = bitbuffer->bits_per_row[0];
+	uint8_t *bytes = bitbuffer->bb[0];
+	double energy, n_imp;
+	double pulsecount;
+	double seconds;
+	data_t *data;
+        char time_str[LOCAL_TIME_BUFLEN];
+
+	if (num_bits < 64 || num_bits > 100) {
+		return 0;
+	}
+
+	// The bit buffer isn't always aligned to the transmitted data, so
+	// search for data start and shift out the bits which aren't part
+	// of the data. The data always starts with 0000 (or 1111 if
+	// gaps/pulses are mixed up).
+	while ((bytes[0] & 0xf0) != 0xf0 && (bytes[0] & 0xf0) != 0x00) {
+		num_bits -= 1;
+		if (num_bits < 64) {
+			return 0;
+		}
+
+		for (unsigned i = 0; i < (num_bits + 7) / 8; ++i) {
+			bytes[i] <<= 1;
+			bytes[i] |= (bytes[i + 1] & 0x80) >> 7;
+		}
+	}
+
+	// Sometimes pulses and gaps are mixed up. If this happens, invert
+	// all bytes to get correct interpretation.
+	if (bytes[0] & 0xf0) {
+		for (unsigned i = 0; i < 11; ++i) {
+			bytes[i] = ~bytes[i];
+		}
+	}
+
+//	unsigned checksum = 0;
+//	for (unsigned i = 0; i < 7; ++i) {
+//		checksum += bytes[i];
+//	}
+//	checksum &= 0xff;
+//	if (checksum != bytes[7]) {
+//		return 0;
+//	}
+
+//	const unsigned VOLTAGES[] = {110, 115, 120, 220, 230, 240, 0};
+
+//	double current_adc = 256 * bytes[4] + bytes[5];
+//	for (unsigned i = 0; VOLTAGES[i] != 0; ++i) {
+//		double power  = (VOLTAGES[i] * current_adc * (1 << bytes[6])) / 32768;
+//		fprintf(stderr, "Power consumption at %u volts: %.2f watts\n", VOLTAGES[i], power);
+//	}
+
+	n_imp = 3200;
+        printf ("\nEfergy Optical\nPulse per KWH default is set to %2f \n", n_imp );
+
+	pulsecount =  bytes[8];
+	seconds = bytes[10];
+//	printf ("\nNumber of Pulses = %2f ", pulsecount);
+//	printf ("\nBytes 0  = %2d", bytes[0]);
+//    printf ("\nBytes 1, Fixed Code 1 = %2d", bytes[1]);
+//    printf ("\nBytes 2, Fixed Code 2 = %2d", bytes[2]);
+//    printf ("\nBytes 3, Fixed Code 3 = %2d", bytes[3]);
+//    printf ("\nBytes 4 = %2d", bytes[4]);
+//    printf ("\nBytes 5 = %2d", bytes[5]);
+//    printf ("\nBytes 6 = %2d", bytes[6]);
+//    printf ("\nBytes 7 = %2d", bytes[7]);
+//    printf ("\nBytes 8, Pulses = %2d", bytes[8]);
+//    printf ("\nBytes 9, = %2d", bytes[9]);
+//    printf ("\nBytes 10, seconds = %2d\n", bytes[10]);
+
```

Comment:
Yes will tidy it up.
On 7 Mar 2016 11:20 p.m., Benjamin Larsson <notifications@github.com> wrote:In src/devices/efergy_optical.c:
> +
> +	pulsecount =  bytes[8];
> +	seconds = bytes[10];
> +//	printf ("\nNumber of Pulses = %2f ", pulsecount);
> +//	printf ("\nBytes 0  = %2d", bytes[0]);
> +//    printf ("\nBytes 1, Fixed Code 1 = %2d", bytes[1]);
> +//    printf ("\nBytes 2, Fixed Code 2 = %2d", bytes[2]);
> +//    printf ("\nBytes 3, Fixed Code 3 = %2d", bytes[3]);
> +//    printf ("\nBytes 4 = %2d", bytes[4]);
> +//    printf ("\nBytes 5 = %2d", bytes[5]);
> +//    printf ("\nBytes 6 = %2d", bytes[6]);
> +//    printf ("\nBytes 7 = %2d", bytes[7]);
> +//    printf ("\nBytes 8, Pulses = %2d", bytes[8]);
> +//    printf ("\nBytes 9, = %2d", bytes[9]);
> +//    printf ("\nBytes 10, seconds = %2d\n", bytes[10]);
> +

Can you remove all the commented code ?

—Reply to this email directly or view it on GitHub.